pallets/stellar-relay/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		// The maximum amount of organizations stored on-chain
+		#[pallet::constant]
+		type OrganizationLimit: Get<u32>;
+
+		// The maximum amount of validators stored on-chain
+		#[pallet::constant]
+		type ValidatorLimit: Get<u32>;
+
+		type WeightInfo: WeightInfo;
+	}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub (super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	// Pallets use events to inform users when important changes are made.
+	// https://docs.substrate.io/v3/runtime/events-and-errors
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {}
+
+	// Errors inform users that something went wrong.
+	#[pallet::error]
+	pub enum Error<T> {
+		Base64DecodeError,
+		BoundedVecCreationFailed,
+		EnvelopeSignedByUnknownValidator,
+		InvalidExternalizedMessages,
+		InvalidEnvelopeSignature,
+		InvalidScpPledge,
+		InvalidTransactionSet,
+		InvalidTransactionXDR,
+		InvalidQuorumSetNotEnoughOrganizations,
+		InvalidQuorumSetNotEnoughValidators,
+		TransactionNotInTransactionSet,
+		TransactionSetHashMismatch,
+		TransactionSetHashCreationFailed,
+		ValidatorLimitExceeded,
+		OrganizationLimitExceeded,
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn organizations)]
+	pub type Organizations<T: Config> =
+		StorageValue<_, BoundedVec<Organization, T::OrganizationLimit>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn validators)]
+	pub type Validators<T: Config> =
+		StorageValue<_, BoundedVec<Validator, T::ValidatorLimit>, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		pub validators: Vec<Validator>,
+		pub organizations: Vec<Organization>,
+		phantom: PhantomData<T>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self { validators: vec![], organizations: vec![], phantom: Default::default() }
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+			let validator_vec =
+				BoundedVec::<Validator, T::ValidatorLimit>::try_from(self.validators.clone());
+			assert!(validator_vec.is_ok());
+			Validators::<T>::put(validator_vec.unwrap());
+
+			let organization_vec = BoundedVec::<Organization, T::OrganizationLimit>::try_from(
+				self.organizations.clone(),
+			);
+			assert!(organization_vec.is_ok());
+			Organizations::<T>::put(organization_vec.unwrap());
+		}
+	}
+
+	// Extrinsics
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// This extrinsic is used to update/replace the current sets of validators and
+		/// organizations
+		#[pallet::weight(<T as Config>::WeightInfo::update_tier_1_validator_set())]
+		pub fn update_tier_1_validator_set(
```

Comment:
Hmm, I think it is highly unlikely that you want to change only the organizations and not the validators. But there might be cases where you only want to update the validators and don't have to touch the organizations. 
IMO splitting this extrinsic is not necessary since even if only one of these two things is required to change, the overhead is just one database write which should be negligible.